src/org/exist/xquery/functions/util/DocumentNameOrId.java

```diff
@@ -163,6 +163,9 @@ public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathExce
                     final int collectionId = absoluteId.and(BigInteger.valueOf(0xFFFFFFFF)).intValue();
                     
                     doc = context.getBroker().getResourceById(collectionId, resourceType, documentId);
+                    if (doc == null) {
```

Comment:
Looks good. I think it would also make sense to add Javadoc on `DBBroker#getResourceById(int, byte, int)` so that it documents that the return type might be null. Would you be able to do that too please?

Reply:
Will do that